### PR TITLE
improved error message for system cmd - config, logs

### LINKED ISSUE #6266: [mesheryctl] improve the error messages shown to users for "mesheryctl system" (1)
**Description**: Improve the error messages in mesheryctl commands from the user's perspective. 
**Commands in scope of this issue**: `mesheryctl system config`, `mesheryctl system logs`,  `mesheryctl system login`, `mesheryctl system logout`, `mesheryctl system check`, `mesheryctl system dashboard`, `mesheryctl system token`

---

We need to make sure of the following things
- There are proper error messages when users
  - Give more arguments to a command than required
  - Misspell a command
  - Do not give the required input via flags or args
  - Use the command without any arguments or flags.
- The error messages being displayed to users are meaningful from user's perspective. And if applicable, a potential fix for the problem should be shown too.  For example
  - "failed to unmarshal response body" could be modified to "couldn't process JSON response from server"
  - "manifest not found" could be modified to "manifest not found. Please provide the path to the app manifest."
  
 --- 
  
   📚 For the **Reference** head over to this [pr](https://github.com/meshery/meshery/pull/5661).

### PR DESCRIPTION
**Pull request by kernelpanic77**
Signed-off-by: kernelpanic77 <shanware.ishan@gmail.com>

**Notes for Reviewers**

This PR fixes #6266 

Added user friendly messages for the following system subcommands:
- [x] config
- [x] logs

I am improving the error messages for dashboard, login, logout and token. Will add more commits to the same PR. Please provide feedback for config and logs subcommands.

Suggestions in cobra does not seem to work for sub sub commands.  For instance, 
running
```
mesheryctl pattern aply  
```
does not suggest "did you mean ? apply".
 
**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Chadha93** [comment]
@kernelpanic77  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
@abdullah1308 will you review?

File: mesheryctl/internal/cli/root/system/config.go [hunk 1]
```diff
@@ -346,7 +346,9 @@ Example: mesheryctl system config eks
 Description: Configure the Kubernetes cluster used by Meshery.`
 
 		if len(args) == 0 {
-			return fmt.Errorf("config name not provided\n\n%v", errMsg)
+			return fmt.Errorf("name of kubernetes cluster to configure meshery not provided\n\n%v", errMsg)
+		} else if len(args) > 1 {
+			return fmt.Errorf("Expected one argument recieved multiple arguments")
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/system/config.go:351]:
```suggestion
			return fmt.Errorf("Expected one argument received multiple arguments")
```

**Review by alphaX86 — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/system/config.go [hunk 2]
```diff
@@ -346,7 +346,9 @@ Example: mesheryctl system config eks
 Description: Configure the Kubernetes cluster used by Meshery.`
 
 		if len(args) == 0 {
-			return fmt.Errorf("config name not provided\n\n%v", errMsg)
+			return fmt.Errorf("name of kubernetes cluster to configure meshery not provided\n\n%v", errMsg)
+		} else if len(args) > 1 {
+			return fmt.Errorf("Expected one argument received multiple arguments")
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/system/config.go:351]:
```suggestion
			return fmt.Errorf("expected one argument received multiple arguments")
```

**Review by alphaX86 — CHANGES_REQUESTED** [review]

**Comment by abdullah1308** [comment]
LGTM

**Review by alphaX86 — DISMISSED** [review]
LGTM

File: mesheryctl/pkg/utils/platform.go [hunk 3]
```diff
@@ -599,7 +599,7 @@ func GetRequiredPods(specifiedPods []string, availablePods []v1core.Pod) ([]stri
 		if index := StringContainedInSlice(sp, availablePodsName); index != -1 {
 			requiredPods = append(requiredPods, availablePodsName[index])
 		} else {
-			return nil, fmt.Errorf("invalid pod \"%s\" specified. Run mesheryctl `system status` to view the available pods", sp)
+			return nil, fmt.Errorf("pod \"%s\" specified, does not exist in the 'meshery' namespace. Please  Run mesheryctl `system status` to view the available pods", sp)
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/platform.go:602]:
```suggestion
			return nil, fmt.Errorf("pod \"%s\" specified, does not exist in the 'meshery' namespace. Please  Run `mesheryctl system status` to view the available pods", sp)
```

File: mesheryctl/internal/cli/root/system/config.go [hunk 4]
```diff
@@ -346,7 +346,9 @@ Example: mesheryctl system config eks
 Description: Configure the Kubernetes cluster used by Meshery.`
 
 		if len(args) == 0 {
-			return fmt.Errorf("config name not provided\n\n%v", errMsg)
+			return fmt.Errorf("name of kubernetes cluster to configure meshery not provided\n\n%v", errMsg)
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/config.go:349]:
```suggestion
			return fmt.Errorf("name of kubernetes cluster to configure Meshery not provided\n\n%v", errMsg)
```

Always capitalize Meshery.

**Review by leecalcote — COMMENTED** [review]
Thanks, @kernelpanic77. 👍

**Review by leecalcote — APPROVED** [review]

**Pull request merged**